isok-agent/src/batch_sender/mod.rs

```diff
@@ -39,6 +41,10 @@ impl JobResult {
     pub(crate) fn set_latency(&mut self, latency: Duration) {
         self.latency = Some(latency);
     }
+
+    pub(crate) fn set_details(&mut self, details: Option<Details>) {
+        self.details = details;
```

Comment:
```suggestion
    pub(crate) fn set_details(&mut self, details: impl Into<Option<Details>>) {
        self.details = details.into();
```

This would allow tot skip wrapping details in `Some` (i.e. `set_details(details)` and `set_details(None)`)

Reply:
Indeed, applied